### ISSUE TITLE
Imports without import clause don't report that the specifier can't end with `.ts`

### ISSUE THREAD

**Issue by Andarist** [microsoft/TypeScript#58725]
### 🔎 Search Terms

import clause ts extension resolution

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

https://www.typescriptlang.org/dev/bug-workbench/?module=1#code/PTAEAEDMEsBsFMB2BDAtvAXKZA6ALgM4BQR8AHgA4D2ATnqAMZWIH3KgC8oA5MtySAgwEKdFgBG+YkWipqdHjmC5C-WfPoBvAL6hINKqkXKp-IA

### 💻 Code

```ts
// @filename: a.ts

export const a = 'a'

// @filename: b.ts

import './a.ts'
import {} from './a.ts'
```

### 🙁 Actual behavior

We get one error in `b.ts`

### 🙂 Expected behavior

We should get errors on both imports in `b.ts`

**Comment by jakebailey** [comment]
These imports are canonically unchecked by design. I personally would love to see this change, but IIRC this is currently intentional behavior, so people can write `import "./App.css";`.

**Comment by Andarist** [comment]
I understand this is something that works as expected for a class of imports. I'm specifically reporting this as an issue with `*.ts` imports. Those have special meaning for TS and it feels weird that this works inconsistently for them here.
